app/Notifications/NewRequestBounty.php

```diff
@@ -42,6 +43,32 @@ public function via(object $notifiable): array
         return ['database'];
     }
 
+    /**
+     * Determine if the notification should be sent.
+     *
+     * @return bool
+     */
+    public function shouldSend(User $notifiable): bool
+    {
+        $targetGroup = 'json_request_groups';
+
+        if ($notifiable->notification?->block_notifications == 1) {
+            return false;
+        }
+
+        if (!$notifiable->notification?->show_request_bounty) {
+            return false;
+        }
+
+        if (\is_array($notifiable->notification->$targetGroup)) {
+            // If the sender's group ID is found in the "Block all notifications from the selected groups" array,
+            // the expression will return false.
+            return !\in_array($this->bounty->user->group->id, $notifiable->notification->$targetGroup, true);
+        }
+
+        return true;
```

Comment:
We should be able to assume that it will always be an array (We made migrations to ensure this a few years ago). Also, it's better to directly access the UserNotification property `json_request_groups`. Also `->group->id` -> `->group_id`. Also remove `return @bool` since it's already indicated in the function signature. This goes for all the other notifications too.

```suggestion
     */
    public function shouldSend(User $notifiable): bool
    {
        if ($notifiable->notification?->block_notifications == 1) {
            return false;
        }

        if (!$notifiable->notification?->show_request_bounty) {
            return false;
        }

        // If the sender's group ID is found in the "Block all notifications from the selected groups" array,
        // the expression will return false.
        return !\in_array($this->bounty->user->group_id, $notifiable->notification->json_request_groups, true);
```

Reply:
Done most of your suggestions here too.

Unfortunately, with your code example, some tests keep failing (notifications are not sent tho they should).
When keeping the if:
```
        if (\is_array($notifiable->notification->json_bon_groups)) {
            // If the sender's group ID is found in the "Block all notifications from the selected groups" array,
            // the expression will return false.
            return !\in_array($this->gift->sender->group_id, $notifiable->notification->json_bon_groups, true);
        }

        return true;
```
Larastan does not like the return true, as it thinks code will be terminated before that, but tests work again.

Reply:
I ended up keeping the check for array. In some cases, the tests failed with something like `argument #2 (haystack) must be of type array, null given`.

Reply:
From a quick glance, it looks to be because the factory is generating a word instead of a json array of ints: https://github.com/HDInnovations/UNIT3D-Community-Edition/blob/f554ae3acb204b705c0ccc48cf50098e8f7d8203/database/factories/UserNotificationFactory.php#L60-L67

Reply:
With this and the fixed factory for `UserNotifications`, I managed to update all the Notifications and got all the tests passing again.